src/runtime/mini_hexagon_dma.h

```diff
@@ -18,17 +18,18 @@ typedef unsigned long addr_t;
 
 typedef unsigned int qurt_size_t;
 typedef unsigned int qurt_mem_pool_t;
+#define ENUM  enum __attribute__((aligned(4)))
```

Comment:
We shouldn't #define symbols like 'ENUM', the chances of collision are high. I think that this should be handled either by:

- just putting the aligned attribute on each enum
- or using (u)int32 to store these values

Reply:
Does the runtime code allow us full C++11 usage? If so, why not just use C++11 enums-with-explicit-sizes?

Reply:
Halide Runtime Doesn't support C++11 Usage so we cannot use enums with explicit sizes

Reply:
@dsharletg : We are Redefining enum to address the enum compatibility issues with Hexagon and host machine. Is it okay if we change the ENUM to HEXAGON_ENUM to address collision concern.

Reply:
It should at least be prefixed with HALIDE_.

I'd prefer that we simply don't use the enum type itself to store the value though, and use uint32 instead, so this isn't necessary.

Reply:
Changed the 'ENUM' to 'HALIDE_HEXAGON_ENUM' temporarily to avoid name collision. Will update it to uint32 when the DMA Driver API Changes come in.